platform/http_request.cpp

```diff
@@ -386,6 +387,12 @@ HttpRequest::~HttpRequest()
 {
 }
 
+HttpRequest::ProgressT const & HttpRequest::Progress() const
+{
+  ASSERT(m_progress.first <= m_progress.second, ());
```

Comment:
Зачем вообще используется прогресс для MemoryHttpRequest, если нам не известен заранее его размер?
По коду, у него валидный только first параметр. Можно из ответа сервера брать Content-Length (если он есть), и -1 выставлять в финальный размер. Но это не маленькое изменение, а главное - не понятно, зачем нужное.

Прогресс от промежуточных запросов в память к метасерверу нигде не должен вообще фигурировать. Если кто-то на него по ошибке подписался - то надо исправлять ошибку именно там.

> On Apr 7, 2016, at 09:48, Илья Гречухин <notifications@github.com> wrote:
> 
> In platform/http_request.cpp <https://github.com/mapsme/omim/pull/2756#discussion_r58825664>:
> 
> > @@ -386,6 +387,12 @@ HttpRequest::~HttpRequest()
> >  {
> >  }
> >  
> > +HttpRequest::ProgressT const & HttpRequest::Progress() const
> > +{
> > +  ASSERT(m_progress.first <= m_progress.second, ());
> Размер не известен в MemoryHttpRequest, предлагаю в том единственном месте где делается m_progress.first += size;
> поступать аналогично и с m_progress.second
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2756/files/193bf11a2e6f8a3ead0d909e7f91c3f351ef2e0a#r58825664>